fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/codec/TestingCodec.java

```diff
@@ -192,7 +207,28 @@ public void files(final SegmentCommitInfo info, final Collection<String> files)
 
     @Override
     public CompoundFormat compoundFormat() {
-        return underlying.compoundFormat();
+        if (allowRandomCompoundFiles) {
+            return new LuceneOptimizedCompoundFormat(((LuceneOptimizedCompoundFormat)underlying.compoundFormat()).compoundFormat) {
```

Comment:
Maybe change the name of the `compoundFormat` field to `underlying` or add a method `getUnderlying()`  to `LuceneOptimizedCompoundFormat` to make this statement easier to follow?